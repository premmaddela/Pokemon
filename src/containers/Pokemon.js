import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { GetPokemon } from '../actions/pokemonActions';
import _ from 'lodash';
import { Card } from 'antd';

const Pokemon = (props) => {
  const pokemonName = props.match.params.pokemon;
  const dispatch = useDispatch();
  const pokemonState = useSelector(state => state.Pokemon);
  
  useEffect(() => {
    dispatch(GetPokemon(pokemonName));
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const RenderData = () => {
    if (!_.isEmpty(pokemonState.data[pokemonName])) {
      const pokeData = pokemonState.data[pokemonName];
      return(
        <div>
        <div className='selectedPoke'>
            <Card bordered={false} style={{ width: 300, backgroundColor: 'lightgray' }}
            cover={<img alt="" src={pokeData.sprites.front_default} />} />
            <h1>{pokemonName.toUpperCase()}</h1>
        </div>
        <div style={{ display: 'inline-flex' }} className='selectedPoke'>
            <Card title='Stats' bordered={false} style={{ width: 300, backgroundColor: 'lightgray' }}>
            {pokeData.stats.map(i => {
              return (
                  <p>{i.stat.name}: {i.base_stat}</p>
              );
            })}
              <p>Height: {pokeData.height}</p>
              <p>Weight: {pokeData.weight}</p>
            </Card>
            <Card title='Abilities' bordered={false} style={{ width: 300, backgroundColor: 'lightgray' }}>
            {pokeData.abilities.map(i => {
              return (
                <p>{i.ability.name}</p>
              )
            })}
            </Card>
        </div>
        </div>
      )
    }

    if (pokemonState.loading) {
      return <p style={{ fontSize: '3rem' }}>Loading...</p>
    }

    if (pokemonState.errorMessage !== '') {
      return <p>{pokemonState.errorMessage}</p>
    }

    return <p>Error getting pokemon</p>
  }

  return(
    <div className={'poke'}>
      {RenderData()}
    </div>
  )
};

export default Pokemon;