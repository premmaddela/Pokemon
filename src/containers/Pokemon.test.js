import TestRenderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux';
import Pokemon from './Pokemon';

describe('renders', () => {
    const initialState = {};
    const mockStore = configureMockStore([]);
    mockStore(initialState);

    const store = mockStore({
        Pokemon: { data: {} }
      });
    const props = {
        match: { params : { pokemon : {} }}
    };

    it('Pokemon', () => {
        const act = TestRenderer;
        const root = act.create(
                <Provider store={store}>
                    <Pokemon {...props} />
                </Provider>
            );
        expect(root).toMatchSnapshot();
    });
});
