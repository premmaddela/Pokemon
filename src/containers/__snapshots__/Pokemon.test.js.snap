// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pokemon 1`] = `
<div
  className="poke"
>
  <p />
</div>
`;
